pages/recommended-git-workflow.qmd

```diff
@@ -2,19 +2,18 @@
 
 Our recommended Git/Github developer workflow for versioning and tagging is as follows:
 
-1. Create a development branch and switch to it, e.g.  
-`git checkout -b feature/dev`
-2. Add your developments, commit and push these changes  
-`git add -A`  
-`git commit -m "my changes"`  
-`git push origin feature/dev`, 
-3. Submit a pull request on Github and get your changes reviewed
+1. Create a development branch and switch to it, e.g. 
+`git checkout -b feature/cool_new_feature`
+2. Add your developments, commit and push these changes.
+``` bash
+git add -A  
+git commit -m "Add <foo> so it can do <bar>"
+git push origin feature/cool_new_feature
+```
+3. Submit a Pull Request (PR) on GitHub and go through the review process.
 4. Once changes are approved, the maintainer decides on the major/minor/patch version increment (e.g. `1.3.0`)
 5. The developer updates to the agreed version in `DESCRIPTION`, e.g.  
-`usethis::use_version(1.3.0)`  
-`git add -A`  
-`git commit -m "incrementing to v1.3.0"`  
-`git push origin feature/dev`
+`usethis::use_version(which = "minor", push = TRUE)`  
```

Comment:
I guess using the argument `minor` automatically adds 0.0.1 to the current version?

Reply:
If you don't specify the version it gives you an interactive prompt. But, not quite, in this usage `"minor"` will increment `x` in `n.x.n`. Other options are `"major"`, (`"minor"`), `"patch"` and `"dev"` (`usethis::use_dev_version()` will also just do the dev version. - [https://usethis.r-lib.org/reference/use_version.html]().

The nice thing with using `usethis` is that it does multiple steps for you - version in DESCRIPTION, a section in NEWS and commits (and optionally pushes too), which is quicker/easier but also makes it less likely for you to forget something.